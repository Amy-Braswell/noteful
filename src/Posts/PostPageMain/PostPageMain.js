import React from 'react'
import ApiContext from '../../ApiContext'
import { findPost } from '../posts-helpers'
import ConnectivityButton from '../../ConnectivityButton/ConnectivityButton'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'


export default class PostPageMain extends React.Component {
  static defaultProps = {
    history: {
      goBack: () => { 
        window.history.back()
       }
    },
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

  handleDeletePost = postId => {
    this.props.history.push(`/`)
  }

  render(){
    const { posts=[] } = this.context
    const { postId } = this.props.match.params
    const post = findPost(posts, postId) || { content: '' }
    return(
      <section className='ItemPageMain'>
        <div className='ItemPageMain__content'>
          <p>Member:{' '}{post.nickname}</p>
          <div className='ItemPageMain__content'>
          {post.content.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>Content:{' '}{para}</p>
          )}
          </div>
          <p>Like Count:{' '}{post.like_count}</p>
          <p>Comment Count:{' '}{post.comment_count}</p>
          <p>Posted:{' '}{post.modified}</p>
          <button
            className='Item__delete'
            type='button'
            onClick={this.handleClickDelete}
          >
            <FontAwesomeIcon icon={['fa', 'trash-alt']} />
            {' '}
            remove
          </button>
        </div>

        <div className='Main__button-container'>
          <ConnectivityButton
            tag='button'
            role='link'
            onClick={() => this.props.history.goBack()}
            className='Nav__back-button'
          >
            <FontAwesomeIcon icon={['fa', 'chevron-left']} />
            <br />
            Back
          </ConnectivityButton>
        </div>

      </section>
    )
  }
}