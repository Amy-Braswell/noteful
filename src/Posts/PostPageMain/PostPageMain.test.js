import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PostPageMain from './PostPageMain'

describe(`PostPageMain component`, () => {
  it('renders a .PostPageMain by default', () => {
    const wrapper = shallow(<PostPageMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip('renders a Post with post prop', () => {
    const props = {
      match: {
        params: {
          postId: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1'
        }
      }
    }
    const context = {
      posts: [{
        id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
        name: `Dogs`,
        modified: `2019-01-03T00:00:00.000Z`,
        // memberid: b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1,
        content: "Corporis accusamus placeat.\n \rUnde."
      }]
    }
    const post = shallow(<PostPageMain {...props} />, context)
      .find('Post')
    expect(toJson(post)).toMatchSnapshot()
  })

  // enzyme doesn't yet support React.createContext
  it.skip(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    const props = {
      match: {
        params: {
          postId: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1'
        }
      }
    }

    const postsContextWithDifferentContent = [
      {
        posts: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            content: "Content with n r.\n \rafter n r.",
          }
        ]
      },
      {
        posts: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            content: "Content with n.\nafter."
          }
        ]
      }
    ]

    postsContextWithDifferentContent.forEach(context => {
      const content = shallow(<PostPageMain {...props} />, context)
        .find('PostPageMain__content')
      expect(toJson(content)).toMatchSnapshot()
    })
  })
})