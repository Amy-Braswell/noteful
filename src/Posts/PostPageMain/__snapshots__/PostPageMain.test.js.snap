// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPageMain component renders a .PostPageMain by default 1`] = `
<section
  className="ItemPageMain"
>
  <NavConnectivityButton
    className="Nav__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fa",
          "chevron-left",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <br />
    Back
  </NavConnectivityButton>
  <div
    className="ItemPageMain__content"
  >
    <p>
      Member:
       
    </p>
    <div
      className="ItemPageMain__content"
    >
      <p
        key="0"
      >
        Content:
         
      </p>
    </div>
    <p>
      Like Count:
       
    </p>
    <p>
      Comment Count:
       
    </p>
    <p>
      Posted:
       
    </p>
    <button
      className="Item__delete"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fa",
            "trash-alt",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      remove
    </button>
  </div>
</section>
`;
